src/crystal/system/unix/user.cr

```diff
@@ -0,0 +1,45 @@
+require "c/pwd"
+
+module Crystal::System::User
+  private def from_struct(pwd)
+    user = String.new(pwd.pw_gecos).split(",").first
+    new(String.new(pwd.pw_name), pwd.pw_uid.to_s, pwd.pw_gid.to_s, user, String.new(pwd.pw_dir), String.new(pwd.pw_shell))
+  end
+
+  def from_username?(username : String)
+    username.check_no_null_byte
+
+    pwd = uninitialized LibC::Passwd
+    pwd_pointer = pointerof(pwd)
+    buf = Bytes.new(1024)
```

Comment:
Tiny optimization: we could use a StaticArray on the stack, instead of allocating HEAP memory, only allocating a dynamic Bytes if 1024 bytes aren't enough:

```crystal
buf = uninitialized UInt8[1024]
```

Reply:
One slight annoyance: I had to go with 

```crystal
initial_buf = uninitialized UInt8[1024]
buf = initial_buf.to_slice
```

since the compiler likes neither the union type (maybe `StaticArray` could be special-cased like `Slice` for C FFI?) nor either of these syntaxes:

```crystal
buf = (uninitialized UInt8[1024]).to_slice
```

```crystal
buf = uninitialized UInt8[1024].to_slice
```